fetch-in-react-practice/client/components/app.jsx

```diff
@@ -0,0 +1,84 @@
+import React from 'react';
+import PageTitle from './page-title';
+import TodoList from './todo-list';
+import TodoForm from './todo-form';
+
+export default class App extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      todos: []
+    };
+    this.addTodo = this.addTodo.bind(this);
+    this.toggleCompleted = this.toggleCompleted.bind(this);
+  }
+
+  componentDidMount() {
+    fetch('/api/todos')
+      .then(res => res.json())
+      .then(data => {
+        this.setState({ todos: data });
+      })
+      .catch(error => console.error(error));
+  }
+
+  // John needs to be nicer
+  // newTodo = { task: "John needs to be nicer", isCompleted: false }
+  // newTodo = { task: "zed", isCompleted: false }
+  addTodo(newTodo) {
+    fetch('api/todos', {
+      method: 'POST',
+      headers: { 'Content-Type': 'application/json' },
+      body: JSON.stringify(newTodo)
+    })
+      .then(res => res.json())
+      .then(data => {
+        const allTodos = this.state.todos.concat(data);
+        this.setState({ todos: allTodos });
+        // this.state.todos = [ {task: 1, description: meh}, {task: 2, description: meh}, {task: 3, description: meh}]
+        // data = { task: "John needs to be nicer", isCompleted: false }
+      })
+      .catch(error => {
+        console.error(error);
+      });
+  }
+
+  toggleCompleted(todoId) {
+    // use Array.prototype.find since ids and indexes normally do not line up
+    const foundTask = this.state.todos.find(todoList => todoList.todoId === todoId);
+
```

Comment:
You need to toggle the `isComplete` here before making the request. The server just saves whatever you send it so as it is now the server is not saving the correct `isComplete` value for your todo